app/assets/stylesheets/components/syntax.scss

```
@@ -34,6 +35,100 @@ div.highlight {
   }
 }
 
+div.highlight {
+  position: relative;
+
+  // this a way to separate scroll of long code & icons in panel
+  .highlight {
+    overflow: auto;
+  }
+}
+
+.highlight__panel {
+  position: absolute;
+  top: 0;
+  right: 0;
+  display: flex;
+  visibility: hidden;
+}
+
+.highlight:hover .highlight__panel {
+  visibility: visible;
+}
+
+.highlight__panel-action {
+  display: flex;
+  overflow: hidden;
+  border-bottom-left-radius: var(--radius);
+
+  & + & {
+    margin-left: 1px;
+    border-bottom-left-radius: 0;
+  }
+}
+
+.highlight-action {
+  display: flex;
+  background: #4f575f;
+  opacity: 0.9;
+
+  &:hover {
+    opacity: 1;
+    cursor: pointer;
+  }
+}
+
+.highlight-action__icon {
+  display: flex;
+}
+
+.highlight-action-icon {
+  --icon-size: calc(var(--su-7) - var(--su-2) * 2);
+  display: block;
+  width: var(--su-7);
+  height: var(--su-7);
+  background-repeat: no-repeat;
+  background-position: center center;
+  background-size: var(--icon-size) var(--icon-size);
+}
+
+.highlight-action-icon--fullscreen {
+  background-image: url(image-src('fullscreen-on.svg'));
+
+  .highlight.is-fullscreen & {
+    background-image: url(image-src('fullscreen-off.svg'));
+  }
+}
+
+.fullscreen-code {
+  position: fixed;
+  top: 0;
+  left: 0;
+  z-index: 9999;
+  width: 100%;
+  height: 100vh;
+  background: var(--syntax-background-color);
+  color: var(--syntax-text-color);
+  overflow: auto;
+  visibility: hidden;
+  transform: scale(0.01);
+  transition: transform 0.4s ease-in-out, visibility 0.4s ease-in-out;
```

Can we replace `0.4s ease-in-out` with `var(--transition-props)` - won't be a huge difference visually, but will definitely speed up the transition (which I think could much faster).

```suggestion
  transition: transform var(--transition-props), visibility var(--transition-props);
```